### PR TITLE
[DataStorage] Initial code to create the configuration.toml file while edge-orchestration is running

### PR DESCRIPTION
**Pull request by t25kim**
Signed-off-by: Taewan Kim <t25.kim@samsung.com>

# Description

This PR is the initial code of TOML for creating and modifying DataStorage's configuration.toml file and its attributes (such as device name, host and others) dynamically. 

Related to #302 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Run TC and compare the result log to the [configs/datastorage/configuration.toml](https://github.com/lf-edge/edge-home-orchestration-go/blob/master/configs/datastorage/configuration.toml) file. 
$ go test -v ./internal/controller/storagemgr/config/

### RESULT
```
=== RUN   TestToml
INFO[2021-05-26T15:51:37+09:00]toml_test.go:41 TestToml
[Clients]

  [Clients.Data]
    Host = "127.0.0.1"
    Port = 48080
    Protocol = "http"
    Timeout = 5000

  [Clients.Metadata]
    Host = "127.0.0.1"
    Port = 48081
    Protocol = "http"
    Timeout = 5000

[Device]
  DataTransform = true
  InitCmd = ""
  InitCmdArgs = ""
  MaxCmdOps = 128
  MaxCmdValueLen = 256
  ProfilesDir = "./res"
  RemoveCmd = ""
  RemoveCmdArgs = ""

[[DeviceList]]
  Description = "RESTful Device"
  Labels = ["rest","json"]
  Name = "datastorage"
  Profile = "datastorage"

  [DeviceList.Protocols]

    [DeviceList.Protocols.other]

[Registry]
  CheckInterval = "10s"
  FailLimit = 3
  FailWaitTime = 10
  Host = "127.0.0.1"
  Port = 8500
  Type = "consul"

[Service]
  AsyncBufferSize = 16
  ConnectionRetries = 20
  EnableAsyncReadings = true
  Host = "127.0.0.1"
  Labels = []
  OpenMsg = "REST device started"
  Port = 49986
  Timeout = 5000

[Writable]
  LogLevel = "DEBUG"
--- PASS: TestToml (0.00s)
PASS
ok      github.com/lf-edge/edge-home-orchestration-go/internal/controller/storagemgr/config     0.001s
```

**Test Configuration**:
* Firmware version: Ubuntu 20.04
* Hardware: x86-64

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: internal/controller/storagemgr/config/toml_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+)
+
+var (
+	log = logmgr.GetInstance()
+)
+
+func TestToml(t *testing.T) {
+	SetWritable("DEBUG")
+	SetService(49986, nil)
+	SetRegistry("127.0.0.1", 8500)
```
Comment (MoonkiHong) [inline internal/controller/storagemgr/config/toml_test.go:32]: @t25kim Just for a check: whether or not it triggers https://github.com/lf-edge/edge-home-orchestration-go/issues/273. Thank you! 😄

**Review by MoonkiHong — COMMENTED** [review]
Thank you for another great contribution. Please let me know your opinion about my comment below! 😁

File: internal/controller/storagemgr/config/toml_test.go (hunk 1 above)
Comment (t25kim) [reply internal/controller/storagemgr/config/toml_test.go:32]: @MoonkiHong Thanks for review! Don't worry. SonarCloud does not check with test files.

Comment (MoonkiHong) [reply internal/controller/storagemgr/config/toml_test.go:32]: @t25kim Confirmed! Thank you!! 😁

**Review by MoonkiHong — APPROVED** [review]
LGTM.

**Review by tdrozdovsky — APPROVED** [review]
Great! LGTM

**Comment by MoonkiHong** [comment]
@suresh-lc PTAL. 😄

**Review by suresh-lc — APPROVED** [review]
Thanks for this support of dynamic toml file creation

**Pull request merged**